src/app/community/community.component.ts

```diff
@@ -40,14 +43,21 @@ export class CommunityComponent implements OnInit {
     this.getcommunitylist();
   }
 
-  deleteCommunity(communityId, communityRev) {
-    const val = confirm('Are you sure you want to delete it?');
-    if (val) {
-      this.couchService.delete('communityregistrationrequests/' + communityId + '?rev=' + communityRev)
-        .then((data) => {
-          this.getcommunitylist();
-        }, (error) => this.message = 'There was a problem deleting this community');
-    }
+  deleteClick(community, index) {
+    // The ... is the spread operator. The below sets deleteItem a copy of the community
+    // object with an additional index property that is the index within the communites array
+    this.deleteItem = { ...community, index: index };
```

Comment:
With [the new shorthand property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer) in ES6 you could shorten it to `this.deleteItem = { ...community, index};`.